linera-chain/src/block_tracker.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeSet;
+
+use custom_debug_derive::Debug;
+use linera_base::{
+    data_types::{Blob, Event, OracleResponse, Timestamp},
+    identifiers::{AccountOwner, BlobType, ChainId},
+};
+use linera_execution::{
+    OutgoingMessage, ResourceController, ResourceTracker, SystemExecutionStateView,
+    TransactionOutcome, TransactionTracker,
+};
+use linera_views::context::Context;
+
+use crate::{
+    chain::EMPTY_BLOCK_SIZE,
+    data_types::{OperationResult, ProposedBlock, Transaction},
+    ChainError, ChainExecutionContext, ExecutionResultExt,
+};
+
+/// Tracks execution of transactions within a block.
+/// Captures the resource policy, produced messages, oracle events.
```

Comment:
```suggestion
/// Captures the resource policy, produced messages, oracle responses and events.
```

(Just to avoid overloading the term "events". :sweat_smile:)

Reply:
Done in [Fix typos](https://github.com/linera-io/linera-protocol/pull/3982/commits/9e15aa7a4b38afcebcb24a8a1ecead5edf8ad988)